added Cart Page

omjadhav85 opened the pull request:
added Cart Page

1.  User can use "Add to Cart" button to add the products in the cart.
2. Cart badge will increase the count after adding new items.
3. User can also move items present in wishlist to his cart directly. If item is already present in cart, its quantity will increase.
4. Also added cart summary to show total cost and discounts applied on cart items.

omjadhav85 committed:
[feat] added Cart page

added Cart page with add to cart functionality

omjadhav85 committed:
[fix] show total items count on cart icon

show total count of items on cart icon badge. Also made some responsive design changes.

omjadhav85 committed:
Update CartPage.css

omjadhav85 committed:
added Go to Products button

added Go to Products button if wishlist is empty

adith-a-danthi commented on src/pages/CartPage/CartPage.js:
```diff
@@ -0,0 +1,30 @@
+import { Footer } from "../../components/Footer/Footer";
+import { NavBar } from "../../components/NavBar/NavBar";
+import { CartProduct } from "../../components/ProductCards/CartProduct";
```
You can make use of barrel imports like you've done with the pages.

omjadhav85 replied:
Hey! Thanks for this suggestion...made the changes

adith-a-danthi commented on src/utils/checkAddedInCart.js:
```diff
@@ -0,0 +1,13 @@
+export const checkAddedInCart = (arr, product) => {
+  if (arr.length === 0) {
+    return false;
+  }
+
+  for (let item of arr) {
+    if (item._id === product._id) {
+      return true;
+    }
+  }
+
+  return false;
```
You can use `array.some()` method to simplify this

omjadhav85 replied:
Cool! Did not know about this one...updated the code using this function. Thanks!!

adith-a-danthi reviewed (commented):
Clean Code! Code split into separate components makes it easy to understand. A couple of suggestions
- Try using ES6+ syntax and methods like `some`, `map`, etc.
- Utility functions can be grouped by component/module instead of a single file for each function.
- You can use barrel imports for components as well

omjadhav85 committed:
[refactor] Added all components to single file

[refactor] Added all components to single file and used ES6 syntax in checkAddedInCart function

omjadhav85 committed:
[feat] added filter logic through homepage

abhij1607 reviewed (commented):
Clean Structured Code || Could have avoided using hardcoded brand values/images in brand.js and could  have passed through props || Good Atomic functions

Apart from the suggestion, everything seems good

prachi-sahani commented on src/utils/checkItemWishlist.js:
```diff
@@ -0,0 +1,13 @@
+    return false;
+  }
+
+  for (let item of arr) {
+    if (item._id === product._id) {
+      return true;
+    }
+  }
```
you may use ES6 array methods

omjadhav85 replied:
Hey! thanks for pointing it out. Used arr.some() instead.

prachi-sahani reviewed (commented):
code is well structured and easy to read.
consider using ES6 practices wherever possible

omjadhav85 committed:
Merge branch 'dev' into cart-page

omjadhav85 commented:
> Clean Structured Code || Could have avoided using hardcoded brand values/images in brand.js and could have passed through props || Good Atomic functions
> 
> Apart from the suggestion, everything seems good

Hi Abhi, thanks for reviewing my PR. I used hardcoded brand names as i have not created a brands array in the backend as it will require some changes in mockbee's backend setup files. Will try to implement it later. Thanks!
